### ISSUE TITLE
EMACS ESS "Package cl is deprecated"

### ISSUE THREAD

**Issue by petersaly** [emacs-ess/ESS#1236]
Dear ESS Developers,

I would like to arise an issue with ESS. When EMACS starts, it gives the following warning message:

'Package cl is deprecated'

This is an issue for me because, as a result, I have a problem with my org-mode initiation. 

I have found that this warning message is sent due to the package ESS. The probable cause is that the package cl has been deprecated and replaced by the package cl-lib  in EMACS 27. However, it seems that ESS still requires the package cl.

Please see also this page with suggested solution:
https://discourse.doomemacs.org/t/warning-at-startup-package-cl-is-deprecated/60

I would like to you to revise this issue and update ESS to eliminate this warning message.

Thank you in advance.
Regards,
Peter

**Comment by meedstrom** [comment]
On a quick search, cl is only loaded in two places, and only if you're on Emacs version 25 or less, because it uses the expression:

     (when (< emacs-major-version 26)
        (require 'cl))

It could be that ESS recently started using some dependency that uses `cl`, or it's just your set-up.  Is your ESS the newest version?  Type `M-x ess-version RET`.  If the output is not `18.10.3snapshot`, it must be very old. I know that [MELPA Stable](https://stable.melpa.org/#/ess) still ships `18.10.2` which is over four years out of date.

**Comment by petersaly** [comment]
Dear meedstrom,

Thank you for your help. Indeed, I used an old version of ESS (v18.10.2) with Emacs 27.1.

I have deleted Emacs 27.1 and installed Emacs 28.1.
I have cloned ESS 18.10.3snapshot from the GitHub.
I moved the ESS folder into my ~/.emacs.d/elpa/ directory.
I have set my .emacs.d/elpa/ diectory default recursively by  addition of the suggestion of EmacsWiki (https://www.emacswiki.org/emacs/LoadPath) into my .emacs.el file.

These steps have solved the issue of the warning message ("Package cl is deprecated"); it has not been sent.

However, after this remedy action, my org-mode still did not work properly. So, I have revised my .emacs.el file. Finally, I have changed the order of the instructions of the org-mode configs written in my .emacs.el, but the actual config commands (e.g., syntax) has not changed at all. As a result, my org-mode has been working properly (e.g., org-capture). I suppose, something went wrong with the initiation loading process. Hence, the warning message sent by duo to the old ESS was probably not the cause of the wrong working of my org-mode.  Unfortunately, the my org-ref does not work, but I am sure this is an an issue that is completely independent from ESS.

I have written my experience here because maybe somebody has similar issue and can find this useful. 

I use Ubuntu 22.04 LTS.

To sum up, after upgrading from ESS 18.10.2  to  ESS 18.10.3snapshot with Emacs 28.1 have solved the issue of the warning message "Package cl is deprecated".

Thank your for your help.
Regards,
Peter
